File: combineSubArray/src/main/kotlin/makeArray.java
import java.util.Arrays;
class Solution {
    public int[] solution(int[] arr, int[][] intervals) {
        int[] answer = {};

        for(int[] inter : intervals){
            for(int i = inter[0]; i <= inter[1]; i++){
                answer = Arrays.copyOf(answer, answer.length + 1);
                answer[answer.length - 1] = arr[i];
            }
        }

        return answer;
    }
}

/**
 * level 0) 배열 만들기 3
 * https://school.programmers.co.kr/learn/courses/30/lessons/181895?language=java
 */

File: combineSubArray/src/main/kotlin/makeArray.kt
class Solution {
    fun solution(arr: IntArray, intervals: Array<IntArray>): IntArray {
        var answer: IntArray = intArrayOf()
        for(inter in intervals){
            for(i in inter[0]..inter[1]){
                answer += arr[i]
            }
        }
        return answer
    }
}
/**
 * level 0) 배열 만들기 3
 * https://school.programmers.co.kr/learn/courses/30/lessons/181895?language=kotlin
 */